hydromt/models/model.py

```diff
@@ -158,67 +193,19 @@ def api(cls) -> Dict:
                 _api.update(getattr(base_cls, "_API", {}))
         return _api
 
-    def _check_get_opt(self, opt):
-        """Check all opt keys and raise sensible error messages if unknown."""
-        for method in opt.keys():
-            m = method.strip("0123456789")
-            meth = self._get_sub_method(m)
-            if not callable(meth):
-                raise ValueError(f'Model {self._NAME} has no method "{method}"')
-        return opt
-
-    def _get_sub_method(self, method_name: str) -> Optional[Callable]:
-        # temporary code until the component logic is added.
-        # now we need this to be able to call region.create
-        obj = self
-        for subojb in method_name.split("."):
-            try:
-                obj = getattr(obj, subojb)
-            except AttributeError:
-                return None
-        func = cast(Callable, obj)
-        return func
-
-    def _run_log_method(self, method, *args, **kwargs):
-        """Log method parameters before running a method."""
-        method = method.strip("0123456789")
-        func = self._get_sub_method(method)
-        if not callable(func):
-            raise ValueError(f'Model {self._NAME} has no method "{method}"')
-        signature = inspect.signature(func)
-        # combine user and default options
-        params = {}
-        for i, (k, v) in enumerate(signature.parameters.items()):
-            if k in ["args", "kwargs"]:
-                if k == "args":
-                    params[k] = args[i:]
-                else:
-                    params.update(**kwargs)
-            else:
-                v = kwargs.get(k, v.default)
-                if len(args) > i:
-                    v = args[i]
-                params[k] = v
-        # log options
-        for k, v in params.items():
-            if v is not inspect._empty:
-                self.logger.info(f"{method}.{k}: {v}")
-        return func(*args, **kwargs)
-
     def build(
         self,
-        region: Optional[dict] = None,
+        *,
+        region: dict[str, Any],
         write: Optional[bool] = True,
-        opt: Optional[dict] = None,
+        steps: Optional[list[dict[str, dict[str, Any]]]] = None,
     ):
-        r"""Single method to build a model from scratch based on settings in `opt`.
+        r"""Single method to build a model from scratch based on settings in `steps`.
 
-        Methods will be run one by one based on the order of appearance in `opt`
-        (configuration file). All model methods are supported including
-        setup\_\*, read\_\* and write\_\* methods.
+        Methods will be run one by one based on the order of appearance in `steps`
+        (configuration file). Model and component methods decorated with @hydromt_step
+        Are allowed to be used in this list.
```

Comment:
Sure, I'll take a stab at it

Reply:
I've updated the docstring as best I could. I'm not sure exactly how we should reshape the API docs for this, do you have an idea? Do you want that to happen this PR as well? because I think that will take longer still.

Reply:
see https://github.com/Deltares/hydromt/issues/852